docs/framework/architecture/ui-library.md

```diff
@@ -439,6 +441,170 @@ dropdownView.bind( 'isEnabled' ).toMany( buttons, 'isEnabled',
+
+To create a modal, use the optional {@link module:ui/dialog/dialog~DialogDefinition#isModal `isModal`} argument of the {@link module:ui/dialog/dialog~Dialog#show `Dialog#show()`} method:
+
+```js
+dialog.show( {
+	isModal: true,
+
+	// The rest of the dialog definition...
+} );
+```
+
+There are different ways to close a modal: either by clicking the "Close" button in the corner (if the [header](#header) is visible), an <kbd>Esc</kbd> key press or one of the [action buttons](#action-buttons).
+
+#### Structure and behaviour
+
+A dialog can have three parts, each of which is optional: the [header](#header) (also used as a drag handler), the [content](#contents) (the body of the dialog) and the [action buttons](#action-buttons) area (a collection of buttons). The order of the parts can not be changed.
+
+##### Header
+
+A header may consist of any combination of three elements:
+
+* the icon,
+* the title,
+* the "Close" button.
+
+The "Close" button is added automatically as long as you provide an icon or a title. The code below shows how to create display a dialog with a header that comes with an icon and a title, but without the "Close" button.
+
+```js
+import { icons } from 'ckeditor5/src/core.js';
+
+// ...
+
+dialog.show( {
+	icon: icons.pencil,
+	title: 'My first dialog',
+	// Do not display the "Close" button.
+	hasCloseButton: false,
+
+	// The rest of the dialog definition...
+} );
+```
+
+##### Content
+
+This part can be a single [view](#views) or a collection of views. They will be displayed directly inside a body of a dialog.
+
+##### Action buttons
+
+The last piece of the dialog is the actions area, where the buttons are displayed. Their behaviour can be fully customized using the `onCreate()` and `onExecute()` callbacks. Below you can find an example of the configuration for the four custom buttons:
+
+* The "OK" button that closes the dialog and has a custom CSS class set,
+* The "Set custom title" button that changes the dialog's title,
+* The "Drag detector" button that changes its state once the dialog was dragged by the user.
+* The "Cancel" button that closes the dialog.
+
+```js
+dialog.show( {
+	actionButtons: [
+		{
+			label: 'OK',
+			class: 'ck-button-action',
+			withText: true,
+			onExecute: () => dialog.hide()
+		},
+		{
+			label: 'Set custom title',
+			withText: true,
+			onExecute: () => {
+				dialog.view!.headerView!.label = 'New title';
+			}
+		},
+		{
+			label: 'Drag detector'
+			withText: true,
+			onCreate: buttonView => {
+				buttonView.bind( 'isEnabled' ).to( dialog.view!, 'wasMoved', wasMoved => !wasMoved );
+			}
+		},
+		{
+			label: 'Cancel',
+			withText: true,
+			onExecute: () => dialog.hide()
+		}
+	]
+} );
+```
+
```

Comment:
I'd add a section talking about `onShow` and `onHide`. `onShow` could be with views actually, because you will usually set up something in the view after it is created, but not always, so maybe it can be in separate section.